### DISCUSSION TITLE
How can you ignore crlf?

### DISCUSSION THREAD

**Discussion by oriongonza** [rome/tools#4384] [Q&A]
We want to add `rome ci` to our ci but we use crlf and the formatter complains. How can we specify a flag to ignore end of line styles?

**Comment by ematipico** [comment]
Can you share a snippet/minimal repro?

**Reply by oriongonza** [reply]
in our ci pipeline we have
`pnpm run ci`
package.json
```json
"ci": "rome ci && tsc --noemit && pnpm test"
```

Git is configured so that it changes `lf` to `crlf` when pushing.

**Reply by ematipico** [reply]
At the moment the formatter used the line ending based on the OS where it runs. There isn't a way to manually change it.

Are you looking for an option in the formatter or automatic way to deduce the line ending set in the git repository?

**Reply by oriongonza** [reply]
I just want to ignore the rule so that it doesn't fail if the line endings are wrong. At least on rome ci.

**Comment by FrancescElies** [comment]
We faced something similar, probably not a common setup what we are having.
```
# .editorconfig

[*]
end_of_line = crlf
```
When running `npx rome format --write src` code gets `lf` line endings, so far so good if one would have [autocrlf](https://docs.github.com/en/get-started/getting-started-with-git/configuring-git-to-handle-line-endings#global-settings-for-line-endings) in git as true, but in our case (this is hard to change for us, we are on a monorepo and would be hard to know which kind of side effects that would have).

So there is a missmatch between what the _editor_ and _rome vscode extension or cli_ does.

@ematipico what would you suggest in this case?
Are line endings OS dependent? I am not familiar with the codebase but I looked at [normalize_newlines](https://github.com/rome/tools/blob/ff023172228587d5277537a5fbac0184303e1118/crates/rome_formatter/src/format_element.rs#L176) but I could not see that changing depending on the OS.

PS: we use rome v12.0.0

**Reply by ematipico** [reply]
Hi!

I was wrong, line endings in our formatter are fixed and they use `lf`.

The code can be found inside the printer: https://github.com/rome/tools/blob/main/crates/rome_formatter/src/printer/mod.rs#L119

The code you shared is used inside template literals and strings.

How could Rome help you? Isn't git enough? Would you need an option to set line endings to `crlf` inside the formatter? Or would you prefer an option that retrieves that value from the git attributes?

**Reply by FrancescElies** [reply]
1. A new option like `endOfLine` would already help, that's probably the simplest to implement I guess.

2. Would it be an option to read `indentStyle` and the new option `endOfLine` from editorconfig? If so that would be great, I see there is [a create](https://crates.io/crates/editorconfig) for that (never used it, not sure if it is any good), but in this case you would have to decide if adding an external dependency is what you want in rome.

3. Can one read that info from git attributes? `git config core.autocrlf` gives only true or false which doesn't tell much and if I understand correctly it still leaves the door open for mismatch between editor and rome.

Ideally I would love 2. (rome understands our editorconfig settings for indent-type, indent-size and end-of-line), but I am ok with 1. to, 1. is probably easier to implement and has no dependencies, what do you think?

Let me know if you think I can help with that to move this forward, i am happy to make a pr with a bit of guidance (started exploring the [idea](https://github.com/rome/tools/compare/main...FrancescElies:tools:cesc/end-of-line?expand=1) already).

**Reply by FrancescElies** [reply]
@ematipico you were right about gitattributes, the info could be retrieved from there.

Atm the following workaround does the job for us.
```
# .gitattributes
*.ts text eol=lf
```
And 
```
# .editorconfig
[*.ts]
end_of_line = lf
```

This two configurations help keep, rome, the editor and git in sync.
